core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpMethodRepoDAO.scala

```diff
@@ -46,28 +48,41 @@ class HttpMethodRepoDAO(baseMethodRepoServiceURL: String, apiPath: String = "",
     Seq((Slash ~ "api" ~ Slash ~ "v1").? / ("configurations" | "methods") / Segment / Segment / Segment)
   )
 
-  private def getAgoraEntity( url: String, userInfo: UserInfo ): Future[Option[AgoraEntity]] = {
-    retry(when500) { () =>
-      pipeline[Option[AgoraEntity]](userInfo) apply Get(url) recover {
-        case notOK: RawlsExceptionWithErrorReport if notOK.errorReport.statusCode.contains(StatusCodes.NotFound) => None
-      }
-    }
-  }
-
   private def postAgoraEntity( url: String, agoraEntity: AgoraEntity, userInfo: UserInfo): Future[AgoraEntity] = {
     retry(when500) { () => pipeline[AgoraEntity](userInfo) apply Post(url, agoraEntity) }
   }
 
   override def getMethodConfig( namespace: String, name: String, version: Int, userInfo: UserInfo ): Future[Option[AgoraEntity]] = {
-    getAgoraEntity(s"${methodRepoServiceURL}/configurations/${namespace}/${name}/${version}",userInfo)
+    getAgoraEntity(s"$agoraServiceURL/configurations/$namespace/$name/$version",userInfo)
   }
 
   override def getMethod( method: MethodRepoMethod, userInfo: UserInfo ): Future[Option[AgoraEntity]] = {
     method match {
       case agoraMethod: AgoraMethod =>
-        getAgoraEntity(s"${methodRepoServiceURL}/methods/${agoraMethod.methodNamespace}/${agoraMethod.methodName}/${agoraMethod.methodVersion}", userInfo)
-      case _ =>
-        throw new RawlsException(s"Method repo '${method.repo.scheme}' not yet supported")
+        getAgoraEntity(s"$agoraServiceURL/methods/${agoraMethod.methodNamespace}/${agoraMethod.methodName}/${agoraMethod.methodVersion}", userInfo)
+      case dockstoreMethod: DockstoreMethod =>
+        getDockstoreMethod(dockstoreMethod) flatMap { response: Option[GA4GHTool] =>
+          response match {
+            case Some(tool) =>
+              // TODO: re-using AgoraEntity feels sketchy to me. It seems to work without any changes, but should we create a DockstoreEntity?
+              Future(Some(AgoraEntity(payload = Some(tool.descriptor), entityType = Some(AgoraEntityType.Workflow))))
+            case None => Future(None)
+          }
+        }
+    }
+  }
+
+  private def getAgoraEntity( url: String, userInfo: UserInfo ): Future[Option[AgoraEntity]] = {
+    retry(when500) { () =>
+      pipeline[Option[AgoraEntity]](userInfo) apply Get(url) recover {
+        case notOK: RawlsExceptionWithErrorReport if notOK.errorReport.statusCode.contains(StatusCodes.NotFound) => None
+      }
+    }
+  }
+
+  private def getDockstoreMethod(method: DockstoreMethod): Future[Option[GA4GHTool]] = {
+    pipeline[Option[GA4GHTool]] apply Get(method.ga4ghDescriptorUrl(dockstoreServiceURL)) recover {
+      case notOK: RawlsExceptionWithErrorReport if notOK.errorReport.statusCode.contains(StatusCodes.NotFound) => None
```

Comment:
Would it be better to do something about the other `case`s here?

Reply:
Making a 404 return `None` is possibly attributable to stale/bad data, but I think anything else is programmer error or a system problem and should throw an exception.

I just copied this handling from `getAgoraEntity`, it makes sense to me though.